snowflake/src/test/java/org/apache/iceberg/snowflake/FakeSnowflakeClient.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.snowflake;
+
+import java.util.List;
+import java.util.Map;
+import org.apache.iceberg.jdbc.UncheckedSQLException;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+
+public class FakeSnowflakeClient implements SnowflakeClient {
+  // In-memory lookup by database/schema/tableName to table metadata.
+  private Map<String, Map<String, Map<String, SnowflakeTableMetadata>>> databases =
+      Maps.newTreeMap();
+  private boolean closed = false;
+
+  public FakeSnowflakeClient() {}
+
+  /**
+   * Also adds parent database/schema if they don't already exist. If the tableName already exists
+   * under the given database/schema, the value is replaced with the provided metadata.
+   */
+  public void addTable(
```

Comment:
it seems that it would probably make sense to use `SnowflakeIdentifier` as a parameter rather than `database` / `schema` / `tableName`

Reply:
Done